server/controllers/admin/roles.js

```diff
@@ -0,0 +1,223 @@
+const db = require('../../lib/db');
+const q = require('q');
+const _ = require('lodash');
+
+module.exports.list = list;
+module.exports.detail = detail;
+module.exports.create = create;
+module.exports.update = update;
+module.exports.remove = remove;
+module.exports.affectPages = affectPages;
+module.exports.affectToUser = affectToUser;
+module.exports.listForUser = listForUser;
+
+function list(req, res, next) {
+  const sql = `
```

Comment:
If you use `FilterParser` here (see #2240), it will make this API much easier to extend in the future.